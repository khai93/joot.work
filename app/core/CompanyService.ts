// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

export type Company = {
    company_id?: number,
    company_name: string,
    logoUrl: URL,
    company_link: URL
}

// CompanyService handles manipluating company data
export interface CompanyService {
    create(company: Company): Promise<void>;
    get(opts: Partial<Company>): Promise<Company | undefined>;
}