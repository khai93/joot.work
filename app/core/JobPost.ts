// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

export type JobPost = {
    name: string,
    description: string,
    company: {
        name: string,
        logoUrl: URL,
        link: URL
    },
    postedDate: Date,
    link: URL
}