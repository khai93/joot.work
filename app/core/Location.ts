// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

export interface Location {
    countryCode: string,
    country: string,
    city: string,
    latitude: string,
    longitude: string,
    state: string
};

