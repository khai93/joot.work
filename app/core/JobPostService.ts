// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

import { Company } from "./CompanyService";
import { GetOptions, UpdateOptions } from "./ServiceOptions";

export type JobPost = {
    job_id?: number,
    job_title: string,
    company_id?: number,
    company?: Partial<Company> | null,
    postedDate: Date | null, // If possible, get a Date object
    formattedDate: string | null, // If Date could not be extracted, get the formatted date from the engine's site
    job_link: URL,
    updated_at?: Date
}

export type SerializedJobPost = ReturnType<typeof serializeJobPost>;

// JobPostService handles the creation of job posts in a database
export interface JobPostService {
    create(post: JobPost, engine: string): Promise<void>;
    createOrUpdateMany(posts: JobPost[], engine: string): Promise<void>;
    get(opts: GetOptions<JobPost>): Promise<JobPost[]>;
    update(opts: UpdateOptions<JobPost>): Promise<void>;
}

/**
 * Serializes a JobPost object into a JSON parsable object
 * @param post 
 */
export const serializeJobPost = (post: JobPost) => ({
    ...post,
    company: {
        ...post.company,
        logoUrl: post.company?.logoUrl?.toString(),
        company_link: post.company?.company_link?.toString()
    },
    postedDate: post.postedDate?.toString() || null, // next.js needs null instead of undefined
    formattedDate: post.formattedDate || null,
    job_link: post.job_link.toString()
});

export const parseJobPost = (post: SerializedJobPost): JobPost => ({
    ...post,
    company: {
        ...post.company,
        logoUrl: new URL(post.company?.logoUrl || "https://placehold.jp/64x64.png"),
        company_link: new URL(post.company?.company_link!)
    },
    postedDate: post.postedDate ? new Date(post.postedDate) : null,
    formattedDate: post.formattedDate || null,
    job_link: new URL(post.job_link)
});