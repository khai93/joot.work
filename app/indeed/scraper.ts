// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

import { JobPost } from "@/core/JobPost";
import { JobScraperService, JobSearchFilter } from "@/core/JobScraperService";
import { injectable } from "inversify";
import { parseFetchResponseHTML, parseFetchResponseText } from "../util/parseFetchResponse";

export type IndeedParseResult = {
    companyBrandingAttributes?: {
        logoUrl?: string,
        headerImageUrl?: string
    },
    company: string,
    companyRating: number,
    companyOverviewLink: string,
    expired: boolean,
    title: string,
    urgentlyHiring: boolean,
    viewJobLink: string,
    remoteLocation: boolean,
    formattedRelativeTime: string
}

@injectable()
export class IndeedJobScraper implements JobScraperService {
    async search(keywords: string[], filters?: JobSearchFilter): Promise<JobPost[]> {
        const requestURL = new URL("https://www.indeed.com/jobs");
        requestURL.searchParams.append("q", keywords.join(" "));
        
        const { _body } = await parseFetchResponseText(await fetch(requestURL.toString(), { 
            headers: {
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246",
                "Host": requestURL.host
            },
            credentials: "same-origin"
        }));

        const output: JobPost[] = [];

        const mainMatch = _body.match(/window\.mosaic\.providerData\["mosaic-provider-jobcards"]={(.*)};/m);

        if (mainMatch == null) {
            throw new Error('Linkedin data regex unexpectedly did not work.');
        }

        for (let result of Array.from(JSON.parse(`{${mainMatch[1]}}`).metaData.mosaicProviderJobCardsModel.results) as IndeedParseResult[]) {
            result = result as any;
            output.push({
                name: result.title,
                description: "",
                company: {
                    name: result.company,
                    link: new URL("https://www.indeed.com" + result.companyOverviewLink),
                    logoUrl: new URL(result.companyBrandingAttributes?.logoUrl || "http://placehold.jp/64x64.png")
                },
                link: new URL("https://www.indeed.com" + result.viewJobLink),
                postedDate: null,
                formattedDate: result.formattedRelativeTime
            });
        }

        return output
    }
}