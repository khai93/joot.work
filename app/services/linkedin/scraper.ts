// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

import { JobPost } from "@/core/JobPost";
import { JobScraperService, JobSearchFilter } from "../../core/JobScraperService";
import { injectable } from "inversify";

import { AxiosInstance } from "axios";

@injectable()
export class LinkedinJobScraper implements JobScraperService {
    constructor(
    ) {}

    search(keywords: string[], filters?: JobSearchFilter): Promise<JobPost[]> {
        throw new Error("Method not implemented.");
    }
}