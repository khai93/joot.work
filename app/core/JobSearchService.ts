// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

import { JobPost } from "./JobPostService";



export type JobSearchFilter = {
    location?: string,
    remoteType?: number // 0 - In-office, 1 - Remote Only, 2 - Hybrid
    page?: number,
    company?: string
}

/**
 * JobSearchService are services that scrapes websites related to searching for jobs
 */
export interface JobSearchService {
    search(keywords: string[], filters?: JobSearchFilter): Promise<JobPost[]>;
}