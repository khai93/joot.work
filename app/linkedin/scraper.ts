// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

import { JobPost } from "@/core/JobPostService";
import { JobScraperService, JobSearchFilter } from "@/core/JobScraperService";
import { injectable } from "inversify";
import { parseFetchResponseHTML } from "../util/parseFetchResponse";

@injectable()
export class LinkedinJobScraper implements JobScraperService {
    constructor() {}

    async search(keywords: string[], filters?: JobSearchFilter): Promise<JobPost[]> {
        const requestURL = new URL('https://www.linkedin.com/jobs/search/');
        requestURL.searchParams.append("keywords", keywords.join(" "));

        const { _document } = await parseFetchResponseHTML(await fetch(requestURL.toString(), { 
            headers: {
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246",
                "Host": requestURL.host
            },
            credentials: "same-origin"
        }));

        const output: JobPost[] = [];
        
        const searchResults = _document.getElementsByClassName("jobs-search__results-list")[0].children;
        
        for (let i = 0;i < searchResults.length;i++) {
            const subtitle = searchResults[i].querySelector(".base-search-card__subtitle");
            if (subtitle?.firstElementChild == null) continue;
            output.push({
                job_title: searchResults[i]!.querySelector(".base-search-card__title")!.textContent!.trim(),
                company: {
                    company_name: subtitle!.firstElementChild!.innerHTML.trim(),
                    company_link: new URL(subtitle!.firstElementChild!.getAttribute("href")!),
                    logoUrl: new URL(searchResults[i]!.querySelector(".artdeco-entity-image")!.getAttribute("data-delayed-url")!)
                },
                postedDate: new Date(searchResults[i]!.querySelector("time")!.getAttribute("datetime")!),
                formattedDate: null,
                job_link: new URL(searchResults[i]!.querySelector(".base-card__full-link")!.getAttribute("href")!)
            });
        }

        return output;
    }
}