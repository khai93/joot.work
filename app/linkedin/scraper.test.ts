// Copyright 2022 Nang Khai.  All rights reserved.
// Use of this source code is governed by a GNU General Public License v3.0
// license that can be found in the LICENSE file.

import { JobPost } from "@/core/JobPost";
import { LinkedinJobScraper } from "./scraper";
import { rest } from "msw";
import { setupServer } from "msw/node";
import fs from "fs/promises";
import path from "path";

const server = setupServer(
    rest.get("https://www.linkedin.com/jobs/search", async (req, res, ctx) => {
        const mockHTMLPage = await fs.readFile(path.resolve(__dirname, "./__mocks__/linkedinSearchMock.html"));
        
        return res(ctx.set('Content-Type', 'text/html'), ctx.body(mockHTMLPage));
    })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('LinkedinJobScraper', () => { 
    describe('#search', () => {
        it('should return correct test data', async () => {
            const scraper = new LinkedinJobScraper();
            expect(await scraper.search(['test'])).toMatchSnapshot();
        });
    })
})