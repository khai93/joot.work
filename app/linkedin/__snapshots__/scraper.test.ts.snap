// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedinJobScraper #search should return correct test data 1`] = `
Array [
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/techoptimum?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQEDBCwDtzOidw/company-logo_100_100/0/1650307219019?e=1661990400&v=beta&t=sfzPO_4882SWKZ8a4bi9ro-JhN_EHOu2aNIdn1Mcyi4",
      "name": "Tech Optimum",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/web-developer-at-tech-optimum-3093685722?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=911jW2odcZucAgPrxZzSQA%3D%3D&position=1&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Web Developer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/caplight-technologies?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQFGMpu_plAuiQ/company-logo_100_100/0/1620340892788?e=1661990400&v=beta&t=R79bSS2bquhmuQdVL52Bh8s_iW13Ch1-xRjMIWsAfGM",
      "name": "Caplight",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/frontend-engineer-at-caplight-3101058134?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=UasOjyxB54DO3md9fyJjHg%3D%3D&position=2&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Frontend engineer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/servicenow?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQEuzVBzTq8wqA/company-logo_100_100/0/1642492123759?e=1661990400&v=beta&t=SUU86Hs4CQo2WswjMIewZ49VWvLFOnyj7JQWwV99cVA",
      "name": "ServiceNow",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/associate-software-engineer-front-end-at-servicenow-3091727672?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=Iv%2FW3AMMynGINGZwrmwulw%3D%3D&position=3&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Associate Software Engineer - Front End",
    "postedDate": 2022-05-25T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/getrevela?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQEfCkhI3DX8eQ/company-logo_100_100/0/1649434526394?e=1661990400&v=beta&t=_XHw_WfRe4CtsNgnxtUlYADxSP-rOkvE1_wNuyG-sEc",
      "name": "Revela",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/junior-front-end-developer-at-revela-3095974085?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=gPINv3avt4Uzgx9Dfjgjgg%3D%3D&position=4&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Junior Front-End Developer",
    "postedDate": 2022-05-27T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/walgreens?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQEAAqRHE5JbYQ/company-logo_100_100/0/1625138284157?e=1661990400&v=beta&t=4LOFoYUyRJULsjBs1Nc5tCv4-kpS5rXf1AJUhYMtx9U",
      "name": "Walgreens",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/associate-software-engineer-at-walgreens-3096325954?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=nlh3pKNBH2sT9qZDncF6Ow%3D%3D&position=5&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Associate Software Engineer",
    "postedDate": 2022-05-01T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/empowertoday?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQEZpfRvav4mKg/company-logo_100_100/0/1644267792319?e=1661990400&v=beta&t=JydONq8GhZ8bp0fTHdWdiPeYNSMZFqZ9DV4mrt1_WXM",
      "name": "Empower",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/web-developer-at-empower-3099499000?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=lohRQ2GAz73wIEpb24DOEg%3D%3D&position=6&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Web Developer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/empowertoday?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQEZpfRvav4mKg/company-logo_100_100/0/1644267792319?e=1661990400&v=beta&t=JydONq8GhZ8bp0fTHdWdiPeYNSMZFqZ9DV4mrt1_WXM",
      "name": "Empower",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/front-end-engineer-at-empower-3099491982?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=McwqZutiSDHkz7kOw6naOQ%3D%3D&position=7&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Front End Engineer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/empowertoday?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQEZpfRvav4mKg/company-logo_100_100/0/1644267792319?e=1661990400&v=beta&t=JydONq8GhZ8bp0fTHdWdiPeYNSMZFqZ9DV4mrt1_WXM",
      "name": "Empower",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/web-developer-at-empower-3099499005?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=CzqrND%2BAg1jQkCNN6ODL5A%3D%3D&position=8&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Web Developer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/kforce?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQEtonVHDHqAZQ/company-logo_100_100/0/1639415774981?e=1661990400&v=beta&t=fUW8UtJpUlb2IAtAc8kLP-6y3t35cPp_cSGRRSYydYw",
      "name": "Kforce Inc",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/front-end-developer-at-kforce-inc-3095049866?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=3cT%2FdzIn9P9uSO48pScl%2Bg%3D%3D&position=9&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Front End Developer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/macguyvertech?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQF3Fs0dxXMukg/company-logo_100_100/0/1625087029143?e=1661990400&v=beta&t=mfUizHNJ3FJvI2Rxyg-5W2jcYr4QZdFmozhUzC3hXp0",
      "name": "MacguyverTech",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/front-end-developer-at-macguyvertech-3098352389?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=oQ%2BnBxZJIOYXe9PsY953IQ%3D%3D&position=10&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Front End Developer",
    "postedDate": 2022-05-27T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/spacelined?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQELlhSXIczyag/company-logo_100_100/0/1649214748041?e=1661990400&v=beta&t=PASmGzMOfTV_3bwpmoWMwJxR2vmsBgen5cRUJJX-SSA",
      "name": "Spacelined",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/web-developer-at-spacelined-3099318046?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=hOfPUPHv%2BqzeObunybvbYw%3D%3D&position=11&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Web Developer",
    "postedDate": 2022-05-29T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/walgreens?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQEAAqRHE5JbYQ/company-logo_100_100/0/1625138284157?e=1661990400&v=beta&t=4LOFoYUyRJULsjBs1Nc5tCv4-kpS5rXf1AJUhYMtx9U",
      "name": "Walgreens",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/associate-software-engineer-at-walgreens-3091035618?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=vKC3aAKTzzVsvQXLR8AnSA%3D%3D&position=12&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Associate Software Engineer",
    "postedDate": 2022-04-28T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/net-inspect?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQHCDEU1U0uoKg/company-logo_100_100/0/1519900679135?e=1661990400&v=beta&t=QnaBLmrXYhGyOgu2SNFdJnDKzHjv8BaNIoW5lA1ptIk",
      "name": "Net-Inspect, LLC",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/junior-front-end-developer-at-net-inspect-llc-3088264791?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=BRhw%2BXN5B722u0eTeK0cJA%3D%3D&position=13&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Junior Front-End Developer",
    "postedDate": 2022-05-26T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/honeywell?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQFvcIh3UnA5zw/company-logo_100_100/0/1535662846602?e=1661990400&v=beta&t=16bcK5RQP6CYNAJVD52sBpHfuw45jANIHnTKSORNoFo",
      "name": "Honeywell",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/front-end-reactjs-developer-at-honeywell-3095273410?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=VQerVVgVaCvZXMR41y9UwQ%3D%3D&position=14&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Front End ReactJS Developer",
    "postedDate": 2022-05-26T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/kforce?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQEtonVHDHqAZQ/company-logo_100_100/0/1639415774981?e=1661990400&v=beta&t=fUW8UtJpUlb2IAtAc8kLP-6y3t35cPp_cSGRRSYydYw",
      "name": "Kforce Inc",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/frontend-developer-react-css-html-at-kforce-inc-3095055027?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=vfZFadYZ7n9jEpZl%2FD2bTA%3D%3D&position=15&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Frontend Developer (React, CSS, HTML)",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/vegan-outreach?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQE3odBavdNZGQ/company-logo_100_100/0/1603315211092?e=1661990400&v=beta&t=czrKhAIm-3buowbZVW7ZHl3_2iJZ7khvV-qo5BOMHgM",
      "name": "Vegan Outreach",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/web-developer-at-vegan-outreach-3095595522?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=Xx%2Bpm6i%2FDC%2F62RY5duqoPw%3D%3D&position=16&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Web Developer",
    "postedDate": 2022-05-27T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/niantic?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQH4GoiewlM7wQ/company-logo_100_100/0/1596236154885?e=1661990400&v=beta&t=3yyDwK5XC8tPlQ7CZ7T6e7El7BwELbZO8mfBzFYD3ug",
      "name": "Niantic, Inc.",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/web-developer-at-niantic-inc-3024132471?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=LfXEsYPFISoc8dEtfNzBAA%3D%3D&position=17&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Web Developer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/actalentservices?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQFqsRl3wX1d2Q/company-logo_100_100/0/1631628816936?e=1661990400&v=beta&t=YWwh0YTDUpvj_oYMC0q3ANhUr6q13ZlBrYcOogVAvd0",
      "name": "Actalent",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/front-end-developer-remote-at-actalent-3102247046?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=eoNGRpwjYqhgTn8g%2BbXJPQ%3D%3D&position=18&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Front End Developer *REMOTE*",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/lockheed-martin?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4D0BAQH-F0QK2LbBzw/company-logo_100_100/0/1519855895078?e=1661990400&v=beta&t=ERUO2PZboTKA0ZLXp0Vp7XOuuBAUIiN3NkE6eBDjll0",
      "name": "Lockheed Martin",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/entry-level-full-stack-developer-at-lockheed-martin-3098610598?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=286%2BLqwQkV83Mr%2FRWmhyZQ%3D%3D&position=19&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Entry Level Full Stack Developer",
    "postedDate": 2022-05-05T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/charterup?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4E0BAQGffxCrEoWqww/company-logo_100_100/0/1580763908944?e=1661990400&v=beta&t=Q51AIdIE4GrYW-EmFatT9cfRcjVPFBQ-j9NMPyvS8AY",
      "name": "CharterUP",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/junior-software-engineer-full-stack-at-charterup-3089158274?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=gm0RpWynDLxc5B%2BL2vY63w%3D%3D&position=20&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Junior Software Engineer, Full-Stack",
    "postedDate": 2022-05-27T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/foncii?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQFDGQ1cHNLYAQ/company-logo_100_100/0/1633904941858?e=1661990400&v=beta&t=KxThP3RGTU62BLlGNbFwQrbXMYkVcqc4uOperI6RWQo",
      "name": "Foncii",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/full-stack-web-developer-at-foncii-3102262017?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=WQnxQlEFMrUgOwowXNYSpQ%3D%3D&position=21&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Full Stack Web Developer",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/trufl-app?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQE95AcpQF2pBA/company-logo_100_100/0/1646593838997?e=1661990400&v=beta&t=LgffaIWx4LavuFfTT1p7Z85F3kCh97pNS4FOW75kLFw",
      "name": "Trufl App",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/junior-web-developer-at-trufl-app-3072017189?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=QXXOuqMbpjLAWAvJWgUWbw%3D%3D&position=22&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Junior Web Developer",
    "postedDate": 2022-05-14T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/charterup?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C4E0BAQGffxCrEoWqww/company-logo_100_100/0/1580763908944?e=1661990400&v=beta&t=Q51AIdIE4GrYW-EmFatT9cfRcjVPFBQ-j9NMPyvS8AY",
      "name": "CharterUP",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/junior-software-engineer-full-stack-at-charterup-3089151801?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=hlRiaNV0SolwuIjerw6Naw%3D%3D&position=23&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Junior Software Engineer, Full-Stack",
    "postedDate": 2022-05-27T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/globallogic?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQHfsFr91ekzrQ/company-logo_100_100/0/1593608530033?e=1661990400&v=beta&t=VK8LuK_qoIOo2_mFNMASIeHFqqUISwC6n4b3_QhL2do",
      "name": "GlobalLogic",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/front-end-engineer-irc155974-at-globallogic-3097401393?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=tOw9jZLtx4y%2B%2B0k6gWrc4Q%3D%3D&position=24&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Front end engineer IRC155974",
    "postedDate": 2022-05-30T00:00:00.000Z,
  },
  Object {
    "company": Object {
      "link": "https://www.linkedin.com/company/best-buy?trk=public_jobs_jserp-result_job-search-card-subtitle",
      "logoUrl": "https://media-exp1.licdn.com/dms/image/C560BAQGS-yyEDG9UaA/company-logo_100_100/0/1652109428206?e=1661990400&v=beta&t=P_jBmg3jDiTKYkhkoI_T_SgAx-I41pBWyxODd70FBYw",
      "name": "Best Buy",
    },
    "description": "",
    "formattedDate": null,
    "link": "https://www.linkedin.com/jobs/view/full-stack-developer-at-best-buy-3093621582?refId=TsAfrqYZ9ZVwpDC5I89QdQ%3D%3D&trackingId=9ViMmlxsX82Je5JW2sZMlA%3D%3D&position=25&pageNum=0&trk=public_jobs_jserp-result_search-card",
    "name": "Full Stack Developer",
    "postedDate": 2022-05-29T00:00:00.000Z,
  },
]
`;
